gtk2_ardour/piano_roll_header.cc

```diff
@@ -443,25 +449,42 @@ PianoRollHeader::on_expose_event (GdkEventExpose* ev)
 
 		}
 
-		/* render the name of which C this is */
-		if (oct_rel == 0) {
-			std::stringstream s;
+		/* if only one note visible render note name vertical (because A#3 don't fit on key) */
+		if (lowest == highest) {
+			string const note_name = ParameterDescriptor::midi_note_name(i);
 			double y = floor(_view.note_to_y(i)) - 0.5f;
 			double note_height = floor(_view.note_to_y(i - 1)) - y;
 
-			int cn = i / 12 - 1;
-			s << "C" << cn;
+			cr->set_font_size(font_size);
+
+			Cairo::TextExtents te;
+			cr->get_text_extents(note_name, te);
+
+			if (is_white_key)
+				cr->set_source_rgb(0.0f, 0.0f, 0.0f);
+			else
+				cr->set_source_rgb(1.0f, 1.0f, 1.0f);
```

Comment:
Have a look at `Gtkmm2ext::contrasting_text_color()` - in general we prefer slightly off-white text (and it may eventually become themeable)

Reply:
Fixed